### PR TITLE
did:key / ed25519 key generation

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Addresses issue https://github.com/digitalcredentials/docs_internal/issues/31

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
(Not sure where the heck the fonts came from; I'll remove those.)

**Comment by kimdhamilton** [comment]
@dekameron22 -- here is the web did:key implementation I mentioned:
- Source: https://github.com/transmute-industries/did-key.js
- Playground: https://did.key.transmute.industries/

Not sure if it provides the necessary info, but I figured we can all discuss here.

Also, for reference, here is the did:key spec: https://w3c-ccg.github.io/did-method-key/

**Comment by kimdhamilton** [comment]
Just catching up on this and to summarize, the main sticking point is random byte generation in react native.

@dmitrizagidulin found RN 3 libraries that seem to be able to do random byte generation, but none of them worked out of the box with the version of RN we are using:
- `react-native-securerandom`
- `expo-random`
- `react-native-virgil-crypto`

@dekameron22 mentioned https://github.com/lyubo/react-native-sodium, which appears to have something for random byte generation, but I'm not sure if it will have the same problems as the other libraries.

Curious about our options here. Does react native sodium work for our RN version? Can we consider changing our RN version to get this working? Or maybe one of the other libraries will update soon?

**Comment by dekameron22** [comment]
Hey guys!
@kimdhamilton @dmitrizagidulin 

Fixed issue with modules here https://github.com/digitalcredentials/cred-wallet/tree/roev/did-token

it was due to `@transmute/did-key-ed25519` was depended on some NodeJS modules which is not installed by default on React Native. I've added them with browserify + babel rewrite module. seems everything is working properly, please confirm on attached screenshot

`{
  "type": "Ed25519VerificationKey2018",
  "id": "#z6MkoSx3492VMZCA4vXQCvnuuJt9nS2Rj4qvjAbnek1SR6gf",
  "controller": "did:key:z6MkoSx3492VMZCA4vXQCvnuuJt9nS2Rj4qvjAbnek1SR6gf",
  "publicKeyBuffer": {
    "type": "Buffer",
    "data": [
      133,
      164,
      209,
      184,
      44,
      211,
      40,
      136,
      55,
      153,
      202,
      36,
      231,
      53,
      54,
      49,
      252,
      56,
      27,
      137,
      140,
      147,
      247,
      75,
      127,
      215,
      238,
      77,
      177,
      91,
      155,
      64
    ]
  },
  "privateKeyBuffer": {
    "type": "Buffer",
    "data": [
      137,
      80,
      65,
      37,
      152,
      46,
      28,
      71,
      212,
      195,
      71,
      188,
      159,
      249,
      38,
      228,
      241,
      102,
      162,
      214,
      66,
      201,
      178,
      149,
      176,
      186,
      245,
      63,
      75,
      229,
      116,
      99,
      133,
      164,
      209,
      184,
      44,
      211,
      40,
      136,
      55,
      153,
      202,
      36,
      231,
      53,
      54,
      49,
      252,
      56,
      27,
      137,
      140,
      147,
      247,
      75,
      127,
      215,
      238,
      77,
      177,
      91,
      155,
      64
    ]
  }
}`

**Comment by dmitrizagidulin** [comment]
@dekameron22 that's very helpful, thank you! I'll rebase this PR on that did-token branch.

**Comment by dekameron22** [comment]
You're welcome! Let me know if I can somehow help you :)

**Comment by dmitrizagidulin** [comment]
@kimdhamilton this PR implements (and documents usage in unit tests):

* Generating `Ed25519` key pairs, either randomly or from deterministic seed.
* Exporting to/loading from the key pairs to string (for storage in React Native keychains)
* Generating `did:key` DIDs from key pairs
* Resolving `did:key` type DID Documents from DIDs.

File: ios/Podfile [hunk 1]
```diff
@@ -5,25 +5,27 @@ platform :ios, '10.0'
...
+
   # Enables Flipper.
   #
   # Note that if you have use_frameworks! enabled, Flipper will not work and
   # you should disable these next few lines.
-  use_flipper!
-  post_install do |installer|
-    flipper_post_install(installer)
-  end
+  #use_flipper!
```
Comment (kimdhamilton) [inline ios/Podfile:27]: what impact will this have?

/cc @dekameron22 in case it affects what you're doing

**Review by kimdhamilton — CHANGES_REQUESTED** [review]
Looks fantastic, I just had 2 comments:
1. Can you remove the .ttf files? I think you said we might not need them. But maybe keep Evil Icons in there because I'm intrigued... 😈
2. Question/comment about the flipper stuff (inline) -- just looking for clarification on whether that might break anything

**Pull request closed without merging**